libs/dh/market-participant/actors/feature-delegation/src/lib/create/dh-delegation-create-modal.component.ts

```diff
@@ -131,10 +132,16 @@ export class DhDelegationCreateModalComponent extends WattTypedModal<DhActorExte
   constructor() {
     super();
 
-    this.gridAreaOptions$.pipe(takeUntilDestroyed()).subscribe((gridAreas) => {
-      if (gridAreas.length === 1) {
-        this.createDelegationForm.controls.gridAreas.setValue([gridAreas[0].value]);
+    effect(() => {
+      this.gridAreaOptions = this.getGridAreaOptions();
+
+      if (this.gridAreaOptions.length === 1) {
+        this.createDelegationForm.controls.gridAreas.setValue([this.gridAreaOptions[0].value]);
       }
+
+      this.createDelegationForm.patchValue({
+        gridAreas: this.gridAreaOptions.map((gridArea) => gridArea.value),
+      });
```

Comment:
question: isn't this `if`/`else`? Seems like `gridAreas` will be set twice if `gridAreaOptions.length` is 1

Reply:
Fixed :-)